compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -15870,6 +15872,7 @@ private STNode parseStatementStartingBracedListFirstMember() {
                 return parseIdentifierRhsInStmtStartingBrace(readonlyKeyword);
             case STRING_LITERAL:
                 STNode key = parseStringLiteral();
+                key = parseBasicLiteral((STToken) key);
```

Comment:
This can break things. Inside the below if-condition,  a `SpecificFieldNode` is created. But for that, `key` is expected to be a token.

Reply:
Let's add a test case for that as well.